osu.Game/Screens/Select/DifficultyRecommender.cs

```diff
@@ -72,6 +84,36 @@ private void calculateRecommendedDifficulties()
             });
         }
 
+        private IEnumerable<RulesetInfo> bestRulesetOrder;
+
+        private IEnumerable<RulesetInfo> getBestRulesetOrder()
+        {
+            bestRulesetOrder ??= recommendedStarDifficulty.OrderByDescending(pair => pair.Value)
+                                                          .Select(pair => pair.Key)
+                                                          .ToList();
```

Comment:
I'm not sure this aggressive caching is necessary, it's sorting a collection of at most 4 items. This isn't going to kill performance, and just ordering every time would get rid of this field.